addons/l10n_ro_edi_stock/models/l10n_ro_edi_stock_etransport_interface.py

```diff
@@ -0,0 +1,640 @@
+                    errors.append(_("This document has not been successfully sent yet because it contains errors."))
+                else:
+                    errors.append(_("This document has not been corrected yet because it contains errors."))
+            case 'sent':
+                errors.append(_("This document has already been successfully sent to anaf."))
+
+        return errors
+
+    # Template Helpers
+
+    def _get_declarant_ref(self) -> str:
+        self.ensure_one()
+        return self.picking_id.name
+
+    def _get_scheduled_date(self):
+        self.ensure_one()
+        return self.picking_id.scheduled_date.date()
+
+    def _get_commercial_partner(self):
+        self.ensure_one()
+        return self.picking_id.partner_id.commercial_partner_id
+
+    def _get_transport_partner(self):
+        self.ensure_one()
+        return self.picking_id.carrier_id.l10n_ro_edi_stock_partner_id
+
+    # Document Helpers
+
+    def _create_document_sending(self, load_id: str, uit: str):
+        self.ensure_one()
+        return self.env['l10n_ro.edi.stock.document'].create({
+            'interface_id': self.id,
+            'state': 'sending',
+            'load_id': load_id,
+            'uit': uit,
+        })
+
+    def _create_document_sending_failed(self, message: str, load_id: str | None = None, uit: str | None = None):
+        self.ensure_one()
+        return self.env['l10n_ro.edi.stock.document'].create({
+            'interface_id': self.id,
+            'state': 'sending_failed',
+            'message': message,
+            'load_id': load_id,
+            'uit': uit,
+        })
+
+    def _create_document_sent(self, load_id: str, uit: str):
+        self.ensure_one()
+        return self.env['l10n_ro.edi.stock.document'].create({
+            'interface_id': self.id,
+            'state': 'sent',
+            'load_id': load_id,
+            'uit': uit,
+        })
+
+    def _get_failed_documents(self):
+        self.ensure_one()
+        return self.l10n_ro_edi_stock_document_ids.filtered(lambda doc: doc.state == 'sending_failed')
+
+    def _get_sending_and_failed_documents(self):
+        self.ensure_one()
+        return self.l10n_ro_edi_stock_document_ids.filtered(lambda doc: doc.state in ('sending_failed', 'sending'))
+
+    def _get_last_sent_document(self):
+        self.ensure_one()
+        sent_documents = self.l10n_ro_edi_stock_document_ids.filtered(lambda doc: doc.state == 'sent').sorted()
+
+        if sent_documents:
+            return sent_documents[0]
+
+    def _get_current_document(self):
+        self.ensure_one()
+        return self.l10n_ro_edi_stock_document_ids.sorted()[0] if self.l10n_ro_edi_stock_document_ids else None
+
+    # Send Logic
+
+    def _send_etransport_document(self, send_type: str):
+        """
+        Send the eTransport document to anaf
+        :param send_type: 'send' (initial sending of document) | 'amend' (correct the already sent document)
+        """
```

Comment:
Also, this function (and the fetch below) should be placed in the document

Reply:
i will move parts of these function to the document model, but creating/deleting documents must stay in the interface, because the interface owns (and therefore is responsible for) the documents. the interface model is similar to the account.move model in the efactura module. they have the same responsibilities regarding the documents